File: codegen/results/v001/haiku/20260407_185054/python/003_fibonacci/solution.py
import sys

N = int(sys.argv[1])

if N < 1:
    sys.exit()

a, b = 1, 1
print(a)

if N >= b:
    print(b)

while True:
    c = a + b
    if c > N:
        break
    print(c)
    a, b = b, c

File: codegen/results/v001/haiku/20260407_185054/typescript/003_fibonacci/solution.ts
const n = parseInt(process.argv[2], 10);

if (n < 1) {
  process.exit(0);
}

let a = 1;
let b = 1;

console.log(a);

while (b <= n) {
  console.log(b);
  [a, b] = [b, a + b];
}
